### PR TITLE
port more rotation stuff to C++

### PR DESCRIPTION
**Pull request by zingale**
## PR summary

This moves fill_rotational_potential and fill_rotational_psi to C++

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-06-24-002/index.html

File: Source/rotation/Rotation.H [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#include "math.H"
+
+#ifndef CASTRO_ROTATION_H
```
Comment (maxpkatz) [inline Source/rotation/Rotation.H:3]: Let's make this `ROTATION_H`

Comment (zingale) [reply Source/rotation/Rotation.H]: done

File: Source/rotation/Rotation.H [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#include "math.H"
+
+#ifndef ROTATION_H
+#define ROTATION_H
+
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
```
Comment (maxpkatz) [inline Source/rotation/Rotation.H:6]: AMREX_INLINE

Comment (zingale) [reply Source/rotation/Rotation.H]: done

**Pull request merged**